CardData.py

```diff
@@ -182,12 +113,14 @@ def create_card_json(gwent_data_helper, patch):
         card['categoryIds'] = []
         categoriesSum = int(template.find('Categories').find('e0').attrib['V']);
         # XML Card category is the sum of all categories of the card
-        for category in sorted(CATEGORIES, reverse=True):
-            if categoriesSum - category >= 0:
-                categoriesSum -= category
-                card['categoryIds'].append(CATEGORIES[category])
-            if categoriesSum == 0:
-                break
+        for category in range(LAST_CATEGORY, 1, -1):
```

Comment:
is this just a bitmask solver?

```python
for category in CATEGORIES
  mask = 1 >> category
  if mask & Category == mask:
    categories.append(category)
```

```python
# perhaps more easily read
for i, bit in enumerate("{0:b}".format(Category))
  if bit == '1'
    categories.append(i)
```

Not that it should change, just find it interesting

Reply:
Would this work in the case of missing numbers? E.g. 45 is not a category.

Reply:
the first example uses the list of known good categories and only checks those
the second example uses the categories from the card (so if the card says 45, then it writes 45)

presumable the second example is sufficient, and we can remove the `CATEGORIES` array

Reply:
Your second example works with just the CATEGORIES check : 
```
for i, bit in enumerate("{0:b}".format(Category)[::-1]):
  if bit == '1' and i in CATEGORIES :
    categories.append(i)```
We need the CATEGORIES array because we can't add a category number that doesn't exist in the translations (actually 45 and 72. I don't even know why they exists).
I checked a little, that doesn't seem to change much about execution time, and it's less easy to read as code, but the use of bits is more logical in this situation (I used the powers of 2 because I'm not good with bit manipulation).